superdesk/io/commands/update_ingest.py

```diff
@@ -104,8 +104,40 @@ def get_provider_rule_set(provider):
 
 
 def get_provider_routing_scheme(provider):
-    if provider.get('routing_scheme'):
-        return superdesk.get_resource_service('routing_schemes').find_one(_id=provider['routing_scheme'], req=None)
+    """Returns the ingests provider's routing scheme configuration.
+
+    If provider has a routing scheme defined (i.e. scheme ID is not None), the
+    scheme is fetched from the database. If not, nothing is returned.
+
+    For all scheme rules that have a reference to a content filter defined,
+    that filter's configuration is fetched from the database as well and
+    embedded into the corresponding scheme rule.
+
+    :param dict provider: ingest provider configuration
+
+    :return: fetched provider's routing scheme configuration (if any)
+    :rtype: dict or None
+    """
+
+    if not provider.get('routing_scheme'):
+        return None
+
+    schemes_service = superdesk.get_resource_service('routing_schemes')
```

Comment:
Wasn't you who was doing the split? :laughing: That's why I asked you for the insight. ;)

If it is a small thing, I would like to do it in the same PR, but if not, it would be more manageable to open a separate task for it IMO - especially considering the fact that this is an already existing problem, and we can't really resolve everything in a single PR. :)